src/OrchardCore.Modules/OrchardCore.Search.Lucene/Migrations.cs

```diff
@@ -166,36 +166,39 @@ public int Create()
 
                     using (var transaction = connection.BeginTransaction(session.Store.Configuration.IsolationLevel))
                     {
-                        var dialect = session.Store.Configuration.SqlDialect;
+                        var dialect = SchemaBuilder.Dialect;
```

Comment:
Here you are in the delegate of a deferred task, not good to keep a reference on the SchemaBuilder of the Migrations.

Anyway in YesSql `SchemaBuilder.Dialect` is initialized with `configuration.SqlDialect`.

Maybe the issue is only due to the missing usage of the below `dialect.QuoteForColumnName()`.

Reply:
updated